Fix timeseries_as_heatmap plot function

mvdbeek opened the pull request:
This fixes:
```
Traceback (most recent call last):
  File "/private/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpn9tl9wf0/job_working_directory/000/2/configs/tmp_r9i0cvx", line 15, in <module>
    sc.pl.dpt_timeseries(
  File "/Users/mvandenb/miniconda3/envs/anndata/lib/python3.8/site-packages/scanpy/plotting/_tools/__init__.py", line 171, in dpt_timeseries
    timeseries_as_heatmap(
  File "/Users/mvandenb/miniconda3/envs/anndata/lib/python3.8/site-packages/scanpy/plotting/_utils.py", line 206, in timeseries_as_heatmap
    pl.colorbar(shrink=0.5)
  File "/Users/mvandenb/miniconda3/envs/anndata/lib/python3.8/site-packages/matplotlib/pyplot.py", line 2188, in colorbar
    raise RuntimeError('No mappable was found to use for colorbar '
RuntimeError: No mappable was found to use for colorbar creation. First define a mappable such as an image (with imshow) or a contour set (with contourf).
```
I see that in other places plt.colobar is used in this module you're
doing the same thing. I believe this broke in
64f04d8.

mvdbeek committed:
Pass generated img to plt.coloarbar

This fixes:
```
Traceback (most recent call last):
  File "/private/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpn9tl9wf0/job_working_directory/000/2/configs/tmp_r9i0cvx", line 15, in <module>
    sc.pl.dpt_timeseries(
  File "/Users/mvandenb/miniconda3/envs/anndata/lib/python3.8/site-packages/scanpy/plotting/_tools/__init__.py", line 171, in dpt_timeseries
    timeseries_as_heatmap(
  File "/Users/mvandenb/miniconda3/envs/anndata/lib/python3.8/site-packages/scanpy/plotting/_utils.py", line 206, in timeseries_as_heatmap
    pl.colorbar(shrink=0.5)
  File "/Users/mvandenb/miniconda3/envs/anndata/lib/python3.8/site-packages/matplotlib/pyplot.py", line 2188, in colorbar
    raise RuntimeError('No mappable was found to use for colorbar '
RuntimeError: No mappable was found to use for colorbar creation. First define a mappable such as an image (with imshow) or a contour set (with contourf).
```
I see that in other places plt.colobar is used in this module you're
doing the same thing. I believe this broke in
https://github.com/theislab/scanpy/commit/64f04d8441d5cfe61da0338312c2742f4ebbff8b.

ivirshup reviewed (changes requested):
Thanks for the bugfix!

Could this get a test? It could be as simple as: 

* Create a small example that fails in current versions that works now
* Add a link to this PR so we know what issue this is testing for

Don't worry about the failing 3.6 build. That's unrelated.

mvdbeek committed:
Add timeseries test

This fails with
```
    def test_timeseries():
        adata = sc.datasets.pbmc68k_reduced()
        sc.pp.neighbors(adata, n_neighbors=5, method='gauss', knn=False)
        sc.tl.diffmap(adata)
        sc.tl.dpt(adata, n_branchings=1, n_dcs=10)
>       sc.pl.dpt_timeseries(adata, as_heatmap=True)

scanpy/tests/test_plotting.py:862:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
scanpy/plotting/_tools/__init__.py:171: in dpt_timeseries
    timeseries_as_heatmap(
scanpy/plotting/_utils.py:206: in timeseries_as_heatmap
    pl.colorbar(shrink=0.5)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

mappable = None, cax = None, ax = None, kw = {'shrink': 0.5}

    def colorbar(mappable=None, cax=None, ax=None, **kw):
        if mappable is None:
            mappable = gci()
            if mappable is None:
>               raise RuntimeError('No mappable was found to use for colorbar '
                                   'creation. First define a mappable such as '
                                   'an image (with imshow) or a contour set ('
                                   'with contourf).')
E               RuntimeError: No mappable was found to use for colorbar creation. First define a mappable such as an image (with imshow) or a contour set (with contourf).

../../miniconda3/envs/anndata/lib/python3.8/site-packages/matplotlib/pyplot.py:2081: RuntimeError
```
Fixed in https://github.com/theislab/scanpy/pull/1654

mvdbeek commented:
I've added a test in https://github.com/theislab/scanpy/pull/1654/commits/189354eb0074140e3f2204d09b02aaa912d13934

ivirshup reviewed (approved):
Great, thanks for the PR! I've just changed how release notes are handled, so I'll manually add this note myself.

Could you comment here with what that release note should be?

ivirshup commented:
@meeseeksdev backport to 1.7.x

mvdbeek commented:
I guess the PR title would be fine ?
